launcher/config.go

```diff
@@ -309,16 +317,24 @@ func newResource(c *Config) *resource.Resource {
 		attributes = append(attributes, label.String(conventions.AttributeServiceVersion, c.ServiceVersion))
 	}
 
-	if hostname, err := os.Hostname(); err == nil {
-		attributes = append(attributes, label.String(conventions.AttributeHostName, hostname))
-	}
-
 	for key, value := range c.resourceAttributes {
 		if len(value) > 0 {
+			if key == conventions.AttributeHostName {
+				hostnameSet = true
+			}
 			attributes = append(attributes, label.String(key, value))
 		}
 	}
 
+	if !hostnameSet {
+		hostname, err := os.Hostname()
+		if err != nil {
+			c.logger.Debugf("host.name not set: %v", err)
```

Comment:
FFTI:

maybe "unable to set host.name: %v" (otherwise it sounds a little like "you were supposed to set host.name")

or even (anticipating a future question from CS): "unable to set host.name (to fix this, you can use OTEL_RESOURCE_ATTRIBUTES="host.name=<your_host_name>"): %v"

Reply:
Good point, I updated message to align with the other configuration error messages